src/Combinatorics/Matroids/matroids.jl

```diff
@@ -0,0 +1,825 @@
+
+Construct a `matroid` from a revlex-basis-encoding-string `rvlx` of rank `r` and size `n`.
+
+# Examples
+```jldoctest
+julia> matroid_from_revlex_basis_encoding("0******0******0***0******0*0**0****", 3, 7)
+Matroid of rank 3 on 7 elements
+```
+"""
+function matroid_from_revlex_basis_encoding(rvlx::String, r::IntegerUnion, n::IntegerUnion)
+    if match(r"[^*0]",rvlx)!=nothing
+	    error("The revlex encoding uses only `*` and `0`")
+    end
+    if length(rvlx)!= binomial(n,r)
+	    error("The length of the string does not match the rank and number of elements")
+    end
+    return Matroid(Polymake.matroid.Matroid(N_ELEMENTS=n, RANK=r, REVLEX_BASIS_ENCODING=rvlx))
+end
+
+@doc Markdown.doc"""
+    matroid_from_bases(B, [n, E])
+
+# Arguments
+- `B::AbstractVector`: The set of bases of the matroid.
+- `n::InterUnion`: The size of the ground set. The ground set will be `{1,..n}` in this case.
+- `E::AbstractVector`: An explicit ground set passed as vector.
+
+Construct a `matroid` with bases `B` on the ground set `E` (which can be the empty set).
+The set `B` is a non-empty collection of subsets of the ground set `E` satisfying an exchange property,
+and the default value for `E` is the set `{1,..n}` for a non-negative value `n`.
+
+See Section 1.2 of [Oxl11](@cite).
+
+# Examples
+To construct a rank two matroid with five bases on four elements you can write:
+```jldoctest
+julia> B = [[1,2],[1,3],[1,4],[2,3],[2,4]];
+
+julia> M = matroid_from_bases(B,4)
+Matroid of rank 2 on 4 elements
+```
+To construct the same matroid on the four elements 1,2,i,j you may write:
+```jldoctest
+julia> M = matroid_from_bases([[1,2],[1,'i'],[1,'j'],[2,'i'],[2,'j']],[1,2,'i','j'])
+Matroid of rank 2 on 4 elements
+```
+"""
+matroid_from_bases(bases::Union{AbstractVector{T},AbstractSet{T}}, nelements::IntegerUnion; check::Bool=true) where T<:GroundsetType = matroid_from_bases(bases,Vector(1:nelements);check=check)
+
+function matroid_from_bases(bases::Union{AbstractVector{T},AbstractSet{T}}, groundset::GroundsetType; check::Bool=true) where T<:GroundsetType
+    if check && length(groundset)!=length(Set(groundset))
+        error("Input is not a valid groundset of a matroid")
+    end
+    if check && !all([e in groundset for S in bases for e in S])
+        error("The bases contain elements that are not in the groundset")
+    end
+    if length(bases)==0
+        error("The cloecction of bases can not be empty")
+    end
+
+    gs2num = create_gs2num(groundset)
+    if length(bases)==1 && length(first(bases))==0
+        M = Polymake.matroid.Matroid(BASES=Polymake.Array{Polymake.Set{Polymake.to_cxx_type(Int)}}(1),N_ELEMENTS=length(groundset))
+    else
+        pm_bases = [[gs2num[i]-1 for i in B] for B in bases]
+        M = Polymake.matroid.Matroid(BASES=pm_bases,N_ELEMENTS=length(groundset))
+        if check && !Polymake.matroid.check_basis_exchange_axiom(M.BASES)
+            error("Input is not a collection of bases")
+        end
+    end
```

Comment:
```suggestion
    pm_bases = Vector{Int}[[gs2num[i]-1 for i in B] for B in bases]
    M = Polymake.matroid.Matroid(BASES=pm_bases,N_ELEMENTS=length(groundset))
    if check && !Polymake.matroid.check_basis_exchange_axiom(M.BASES)
        error("Input is not a collection of bases")
    end
```
Just make sure there is no `Any[]` in the input then there is no need for such special cases.

To create non-empty arrays without giving the elements please use the undef-initializer (just like for julia arrays):
```julia
julia> Polymake.Array{Polymake.Set{Int}}(undef,1)
pm::Array<pm::Set<long, pm::operations::cmp> >
{}
```
(this does work fine on macos and linux)

Reply:
I am not sure if we can avoid `Any[]` (representing the empty set) in the input

Reply:
I was referring to the input for polymake, the conversion to `pm_bases` where I added a `Vector{Int}` should take care of the conversion `Any[]` to `Int64[]`.

Reply:
I see